src/librustdoc/html/render/print_item.rs

```diff
@@ -430,29 +431,44 @@ fn extra_info_tags(item: &clean::Item, parent: &clean::Item, tcx: TyCtxt<'_>) ->
 }
 
 fn item_function(w: &mut Buffer, cx: &Context<'_>, it: &clean::Item, f: &clean::Function) {
-    let header_len = format!(
-        "{}{}{}{}{:#}fn {}{:#}",
-        it.visibility.print_with_space(it.def_id, cx),
-        f.header.constness.print_with_space(),
-        f.header.asyncness.print_with_space(),
-        f.header.unsafety.print_with_space(),
-        print_abi_with_space(f.header.abi),
-        it.name.as_ref().unwrap(),
-        f.generics.print(cx),
-    )
-    .len();
+    let vis = it.visibility.print_with_space(it.def_id, cx).to_string();
+    let mut constness = f.header.constness.print_with_space();
+    let asyncness = f.header.asyncness.print_with_space();
+    let unsafety = f.header.unsafety.print_with_space();
+    let abi = print_abi_with_space(f.header.abi).to_string();
+    let name = it.name.as_ref().unwrap();
+
+    if !constness.is_empty() {
+        if let Some(ConstStability { level: StabilityLevel::Unstable { .. }, .. }) =
+            it.const_stability(cx.tcx())
+        {
+            constness = "";
+        }
+    }
+
+    let generics_len = format!("{:#}", f.generics.print(cx)).len();
+    let header_len = "fn ".len()
+        + vis.len()
+        + constness.len()
+        + asyncness.len()
+        + unsafety.len()
+        + abi.len()
+        + name.as_str().len()
+        + generics_len;
+
     w.write_str("<pre class=\"rust fn\">");
     render_attributes_in_pre(w, it, "");
+    w.reserve(header_len);
     write!(
         w,
-        "{vis}{constness}{asyncness}{unsafety}{abi}fn \
+        "{}{}{}{}{}fn \
```

Comment:
Why did you change this? I think it was easier to read before.

Reply:
To match `html/render/mod.rs`.

https://github.com/rust-lang/rust/blob/03b845a41f1666654ecffcbaa6a582170ec0ed8d/src/librustdoc/html/render/mod.rs#L898-L934

The length calculation and the printing would reuse most parts instead of creating a temporary string solely to calculate the length. But it might be better to change `{}` into names `{vis}`, etc.

Reply:
> But it might be better to change {} into names {vis}, etc.

Right, this is what I meant, the rest of the changes are fine :)